syncer/syncer.go

```diff
@@ -1226,41 +1227,46 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		case *replication.RotateEvent:
 			err = s.handleRotateEvent(ev, ec)
 			if err != nil {
-				return terror.Annotatef(err, "current pos %s", currentPos)
+				return terror.Annotatef(err, "current location %s", currentLocation)
 			}
 		case *replication.RowsEvent:
 			err = s.handleRowsEvent(ev, ec)
 			if err != nil {
-				return terror.Annotatef(err, "current pos %s", currentPos)
+				return terror.Annotatef(err, "current location %s", currentLocation)
 			}
 		case *replication.QueryEvent:
 			err = s.handleQueryEvent(ev, ec)
 			if err != nil {
-				return terror.Annotatef(err, "current pos %s", currentPos)
+				return terror.Annotatef(err, "current location %s", currentLocation)
 			}
 		case *replication.XIDEvent:
 			if shardingReSync != nil {
-				shardingReSync.currPos.Pos = e.Header.LogPos
-				lastPos = shardingReSync.currPos
-				if binlog.ComparePosition(shardingReSync.currPos, shardingReSync.latestPos) >= 0 {
+				shardingReSync.currLocation.Position.Pos = e.Header.LogPos
+				shardingReSync.currLocation.GTIDSet.Set(ev.GSet)
+
+				// only need compare binlog position?
+				lastLocation = shardingReSync.currLocation.Clone()
+				if binlog.CompareLocation(shardingReSync.currLocation, shardingReSync.latestLocation) >= 0 {
 					s.tctx.L().Info("re-replicate shard group was completed", zap.String("event", "XID"), zap.Reflect("re-shard", shardingReSync))
 					err = closeShardingResync()
 					if err != nil {
-						return terror.Annotatef(err, "shard group current pos %s", shardingReSync.currPos)
+						return terror.Annotatef(err, "shard group current location %s", shardingReSync.currLocation)
 					}
 					continue
 				}
 			}
 
 			latestOp = xid
-			currentPos.Pos = e.Header.LogPos
-			s.tctx.L().Debug("", zap.String("event", "XID"), zap.Stringer("last position", lastPos), log.WrapStringerField("position", currentPos), log.WrapStringerField("gtid set", ev.GSet))
-			lastPos.Pos = e.Header.LogPos // update lastPos
+			currentLocation.Position.Pos = e.Header.LogPos
+			currentLocation.GTIDSet.Set(ev.GSet)
+			s.tctx.L().Debug("", zap.String("event", "XID"), zap.Stringer("last location", lastLocation), log.WrapStringerField("location", currentLocation), log.WrapStringerField("gtid set", ev.GSet))
```

Comment:
Can `, log.WrapStringerField("gtid set", ev.GSet)` be removed now? If so, ditto for other places.

Reply:
removed